### PR TITLE
Ass Blast Usa

### PR DESCRIPTION
**Pull request by karmaisblackandbluepilled**
Adds farting, butts, buttflowers, buttshoes, fartium, Bible gibbing.
There's currently two issues, the fart emote happens twice due to a fix that TCEO gave me and the butt getting blasted off creates a runtime due to a signal? I think. I don't know how to fix these issues.

### CODE REVIEW

File: code/modules/mob/living/carbon/emote.dm [hunk 1]
```diff
@@ -1,6 +1,268 @@
...
+				playsound(user, fartsound, 100, 1, 5)
+		if(blowass)
+			B.Remove(user)
+			B.forceMove(get_turf(user))
+			new bloodkind(user.loc)
+			user.nutrition = max(user.nutrition - rand(5, 20), NUTRITION_LEVEL_STARVING)
+			user.visible_message("<span class='warning'><b>[user]</b> blows their ass off!</span>", "<span class='warning'>Holy shit, your butt flies off in an arc!</span>")
+		else
+			user.nutrition = max(user.nutrition - rand(2, 10), NUTRITION_LEVEL_STARVING)
+		..()
```
Comment (TheChosenEvilOne) [inline code/modules/mob/living/carbon/emote.dm]:
```suggestion
```

File: code/modules/mob/living/carbon/emote.dm [hunk 2]
```diff
@@ -1,6 +1,268 @@
...
+					var/new_dir
+					switch(user.dir)
+						if(1)
+							new_dir = 2
+						if(2)
+							new_dir = 1
+						if(4)
+							new_dir = 8
+						if(8)
+							new_dir = 4
```
Comment (TheChosenEvilOne) [inline code/modules/mob/living/carbon/emote.dm]:
```suggestion
					var/new_dir = REVERSE_DIR(user.dir)
```

File: code/datums/keybinding/emote.dm [hunk 3]
```diff
@@ -16,3 +16,15 @@
...
+	hotkey_keys = list("CtrlS")
+	name = "scream"
+	full_name = "Scream"
+	description = "REEEEEEEEEEEEEEEEEEEEEEEE!"
+
+/datum/keybinding/emote/fart
+	hotkey_keys = list("CtrlF")
+	name = "fart"
+	full_name = "Fart"
+	description = "Uh oh, stinky!"
```
Comment (TheChosenEvilOne) [inline code/datums/keybinding/emote.dm]:
```suggestion
```
these are not needed AFAIK

File: code/modules/surgery/organs/ass.dm [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+/obj/item/organ/butt/on_life()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	if(owner && STR)
+		var/list/STR_contents = STR.contents()
+		for(var/obj/item/I in STR_contents)
+			if(get_sharpness(I) == SHARP_EDGED || get_sharpness(I) == SHARP_POINTY)
+				owner.bleed(4)
+
+/obj/item/organ/butt/attackby(var/obj/item/W, mob/user as mob, params) // copypasting bot manufucturing process, im a lazy fuck
```
Comment (TheChosenEvilOne) [inline code/modules/surgery/organs/ass.dm]:
```suggestion
/obj/item/organ/butt/attackby(obj/item/W, mob/living/user, params) // copypasting bot manufucturing process, im a lazy fuck
```
linter complaining about this, also I don't see a reason for the `mob/user as mob` so

File: code/_onclick/click.dm [hunk 5]
```diff
@@ -214,10 +214,16 @@
 			if (!target.loc)
 				continue
 
+			if(reach_organs && iscarbon(src))
+				var/mob/living/carbon/carbon_src = src
+				if(target in carbon_src.bodyparts || target in carbon_src.internal_organs)
```
Comment (TheChosenEvilOne) [inline code/_onclick/click.dm]: linter screaming about this one
```suggestion
				if((target in carbon_src.bodyparts) || (target in carbon_src.internal_organs))
```

**Comment by karmaisblackandbluepilled** [comment]
don't merge this

**Comment by karmaisblackandbluepilled** [comment]
something here makes bags unusable which isn't that good

**Comment by karmaisblackandbluepilled** [comment]
ok i fixed it time to uhh got dam pussy fart shit nREDACTED push it

File: code/modules/surgery/organs/ass.dm [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+		to_chat(user, "<span class='notice'>You add the robot arm to the butt and... What?</span>")
+		user.dropItemToGround(src)
+		qdel(src)
+
+/obj/item/organ/butt/throw_impact(atom/hit_atom)
+	..()
+	playsound(src, 'sound/effects/fart.ogg', 50, 1, 5)
+
+/mob/living/carbon/proc/regeneratebutt()
+	if(!getorganslot("butt"))
```
Comment (TheChosenEvilOne) [inline code/modules/surgery/organs/ass.dm]:
```suggestion
	if(!getorganslot(ORGAN_SLOT_BUTT))
```

File: code/modules/surgery/organs/organ_internal.dm [hunk 7]
```diff
@@ -248,6 +248,14 @@
 			ears.Insert(src)
 		ears.setOrganDamage(0)
 
+		if(!getorganslot("butt"))
```
Comment (TheChosenEvilOne) [inline code/modules/surgery/organs/organ_internal.dm]:
```suggestion
		if(!getorganslot(ORGAN_SLOT_BUTT))
```

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 8]
```diff
@@ -259,6 +259,23 @@
...
+	name = "Fartium"
+	description = "A chemical compound that promotes concentrated production of gas in your groin area."
+	color = "#8A4B08" // rgb: 138, 75, 8
+
+/datum/reagent/drug/fartium/on_mob_life(mob/living/M)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		var/obj/item/organ/butt/B = locate() in H.internal_organs
+		if(prob(7))
+			if(B)
```
Comment (TheChosenEvilOne) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]:
```suggestion
		if(prob(7))
			if(H.getorganslot(ORGAN_SLOT_BUTT))
```

**Review by TheChosenEvilOne — CHANGES_REQUESTED** [review]
Alright, I don't really see anything wrong other than some minor style issues after these which can be ignored.

**Review by TheChosenEvilOne — APPROVED** [review]

**Comment by TheChosenEvilOne** [comment]
alright, I don't see anything that horribly wrong.

**Pull request merged**